024-file-tokens.md

```diff
@@ -0,0 +1,199 @@
+# File tokens
+
+- Implementation Owner: @lohanidamodar
+- Start Date: 20-06-2023
+- Target Date: N/A
+- Appwrite Issue:
+  [Is this RFC inspired by an issue in appwrite](https://github.com/appwrite/appwrite/issues/)
+
+## Summary
+
+[summary]: #summary
+
+<!-- Brief explanation of the proposed contribution. Write your answer below. -->
+
+Accessing files currently requires either files to be public or need user session with user that have access permission. It's mainly painful for image previews that we want to show publicly. For this reason, we want to introduce file tokens. The tokens will be linked to specific file and will be passed in URL for previews and downloads to provide access.
```

Comment:
We could also mention how this is problematic with some browsers that don't supports Appwrite's secure cookies.